ci/svd2rust-regress/src/diff.rs

```diff
@@ -0,0 +1,311 @@
+                        command = std::process::Command::new(pager.next().unwrap());
+                        command.args(pager);
+                    } else {
+                        command = std::process::Command::new("git");
+                        command.env("GIT_PAGER", pager);
+                    }
+                } else {
+                    command = std::process::Command::new("git");
+                    command.arg("--no-pager");
+                }
+                if !self.use_pager_directly {
+                    command.args(["diff", "--no-index", "--minimal"]);
+                }
+                command
+                    .args([&*baseline.0, &*current.0])
+                    .status()
+                    .with_context(|| "couldn't run diff")
+                    .map(|_| ())
+            }
+            DiffingMode::Semver => std::process::Command::new("cargo")
+                .args(["semver-checks", "check-release"])
+                .arg("--baseline-root")
+                .arg(baseline.0)
+                .arg("--manifest-path")
+                .arg(current.0.join("Cargo.toml"))
+                .status()
+                .with_context(|| "couldn't run semver-checks")
+                .map(|_| ()),
+        }
+    }
+
+    pub fn make_case(&self, opts: &Opts) -> Result<[(PathBuf, Vec<PathBuf>); 2], anyhow::Error> {
+        let [(baseline_bin, baseline_cmd), (current_bin, current_cmd)] = self
+            .svd2rust_setup(opts)
+            .with_context(|| "couldn't setup svd2rust")?;
+        let tests = crate::tests::tests(Some(opts.test_cases.as_ref()))
+            .with_context(|| "no tests found")?;
+
+        let tests = tests
+            .iter()
+            .filter(|t| {
+                if let Some(ref arch) = self.arch {
+                    arch.to_ascii_lowercase()
+                        .eq_ignore_ascii_case(&t.arch.to_string())
+                } else {
+                    true
+                }
+            })
+            // selected manufacturer?
+            .filter(|t| {
+                if let Some(ref mfgr) = self.mfgr {
+                    mfgr.to_ascii_lowercase()
+                        .eq_ignore_ascii_case(&t.mfgr.to_string().to_ascii_lowercase())
+                } else {
+                    true
+                }
+            })
+            .filter(|t| {
+                if self.chip.is_empty() {
+                    false
+                } else {
+                    self.chip.iter().any(|c| {
+                        wildmatch::WildMatch::new(&c.to_ascii_lowercase())
+                            .matches(&t.chip.to_ascii_lowercase())
+                    })
+                }
+            })
+            .collect::<Vec<_>>();
+        let test = match (tests.len(), self.sub) {
+            (1, _) => tests[0],
+            (1.., Some(DiffingMode::Pr)) => tests
+                .iter()
+                .find(|t| t.chip == "STM32F401")
```

Comment:
I basically want to make it easy to generate a diff with

/ci diff pr

that would show the diff of the pr

so instead of always having to do

/ci diff -c AGoodChip

Reply:
ok, should we use https://stm32-rs.github.io/stm32-rs/stm32f103.svd.patched then maybe?

Reply:
fixed!